concent_api/core/message_handlers.py

```diff
@@ -164,23 +164,42 @@ def handle_send_ack_report_computed_task(client_message):
 
 
 def handle_send_reject_report_computed_task(client_message):
+    # Validate if task_to_compute is valid instance of TaskToCompute.
     task_to_compute = client_message.task_to_compute
-
-    validate_golem_message_reject(client_message.cannot_compute_task)
     validate_task_to_compute(task_to_compute)
-    validate_task_to_compute(client_message.cannot_compute_task.task_to_compute)
 
     provider_public_key = task_to_compute.provider_public_key
     requestor_public_key = task_to_compute.requestor_public_key
 
+    # Validate if TaskToCompute signed by the requestor.
     validate_golem_message_signed_with_key(
         task_to_compute,
         requestor_public_key,
     )
-    validate_golem_message_signed_with_key(
-        client_message.cannot_compute_task.task_to_compute,
-        requestor_public_key,
-    )
+
+    tasks_to_compute = [task_to_compute]
+
+    # If reason is GotMessageCannotComputeTask,
+    # cannot_compute_task is instance of CannotComputeTask signed by the provider.
+    if client_message.reason == message.RejectReportComputedTask.REASON.GotMessageCannotComputeTask:
+        validate_golem_message_reject(client_message.cannot_compute_task)
```

Comment:
This validation is broken.
1. 
```python
    if isinstance(golem_message, message.CannotComputeTask):
        validate_id_value(golem_message.task_to_compute.compute_task_def['task_id'], 'task_id')
```
is redundant to `validate_task_to_compute` (the check above is included in the latter)

2.
```python
if isinstance(golem_message, (message.TaskToCompute, message.TaskFailure)):
    if golem_message.compute_task_def['task_id'] == '':
        raise Http400("task_id cannot be blank.")
```


`TaskFailure` does not contain `compute_task_def` directly, once passed to verification `TaskFailure` msg would produce `AttributeError`

3. 
```python
  golem_message.compute_task_def['deadline'] = validate_int_value(golem_message.compute_task_def['deadline'])
```
is also redundant to `validate_task_to_compute`

4. 
I am not sure if we shouldn't also check weather `CannotComputeTask` and `TaskFailure` are simultaneously present int he message https://github.com/golemfactory/concent/issues/286#issuecomment-382278207